File: Suspension matlab/Suspension matlab raw/Natural_frequency/aero_spring.m
clc; clear; close all;
g = 9.81;

%% data
m = 300;                 % 直量kg
a = 1.6;          % 最大加速度m/s^2
h = 0.28;                % 重心高m
L = 1.55;                % 軸距m
%% 計算
% ===== sweep allowable deformation =====
theta = deg2rad(1): deg2rad(1): deg2rad(30);
s = L/2*theta;

F = m * a * g * h/(L/2);

% ===== pitch stiffness =====
K = F./s/2;

% ===== plot =====
figure;
plot( K,theta, 'LineWidth', 2);
grid on;

xlabel('K');
ylabel('theta_allowable');
title('Pitch Stiffness Requirement vs Allowable Deformation');

File: Suspension matlab/Natural_frequency/aero_spring/aero_spring.py
from pathlib import Path
import sys

import matplotlib.pyplot as plt
import numpy as np

ROOT = next(p for p in Path(__file__).resolve().parents if (p / "suspension_utils.py").exists())
sys.path.insert(0, str(ROOT))

from suspension_utils import load_data, show_plots


def main() -> None:
    p = load_data(Path(__file__).with_name("data.csv"))
    g = p["aero_g"]
    m = p["aero_m"]
    a = p["aero_a"]
    h = p["aero_h"]
    length = p["aero_L"]
    theta = np.asarray(p["aero_theta"], dtype=float)

    s = length / 2 * theta
    force = m * a * g * h / (length / 2)
    stiffness = force / s / 2

    plt.figure()
    plt.plot(stiffness, theta, linewidth=2)
    plt.grid(True)
    plt.xlabel("K")
    plt.ylabel("theta_allowable")
    plt.title("Pitch Stiffness Requirement vs Allowable Deformation")
    show_plots()


if __name__ == "__main__":
    main()
